lib/command_runner.dart

```diff
@@ -108,26 +108,26 @@ class CommandRunner<T> {
   ///
   /// This always returns a [Future] in case the command is asynchronous. The
   /// [Future] will throw a [UsageException] if [args] was invalid.
-  Future<T> run(Iterable<String> args) =>
-      Future.sync(() => runCommand(parse(args)));
+  Future<T?> run(Iterable<String> args) =>
```

Comment:
I'm worried about this `T?` but I'm not sure if there is much we can do about it. It looks like we rely on being able to return a default for when `--help` is used. We could also consider making a required constructor argument of type `T` for the default value.

Reply:
Ya I don't think we can really do much here, I wouldn't want to force some arbitrary return value - this would end up requiring a lot of people to pass an explicit `null` for that I think since they don't even use these return values.